lib/offline_repository/repository.dart

```diff
@@ -0,0 +1,68 @@
+import 'dart:convert';
+import 'package:api_client/models/model.dart';
+import 'package:api_client/offline_repository/db_model_factory.dart';
+import 'package:api_client/offline_repository/exceptions/not_found.dart';
+import 'package:api_client/offline_repository/provider.dart';
+import 'package:api_client/offline_repository/repository_interface.dart';
+import 'package:sqflite/sqflite.dart';
+
+/// Implementation of offline repository
+class OfflineRepository implements IOfflineRepository<Model> {
+
+  /// constructor
+  OfflineRepository(this._tableName) {
+    init();
+  }
+
+  /// Initial database connection
+  Future<Database> init() async {
+    _db = await DbProvider.instance.database;
+    return _db;
+  }
+
+  final String _tableName;
+
+  Database _db;
+
+  @override
+  Future<Model> insert(Model model) {
```

Comment:
Is there a reason the methods insert(), update(), delete(), and all are not implemented?

Reply:
Yes https://github.com/aau-giraf/api_client/pull/46#issue-407395781

Reply:
What is the reason then? It's not really clear from the description.

Reply:
The methods should and will be implemented, but are not required for the other groups with offline issues to begin their work. That's why this PR was made without the methods has been implemented. :)